Loaded module: cuda/11.6
Loaded dependency [python3/3.11.4]: gcc/12.3.0-binutils-2.40
Loaded dependency [python3/3.11.4]: sqlite3/3.42.0
Loaded module: python3/3.11.4

Loading python3/3.11.4
  Loading requirement: gcc/12.3.0-binutils-2.40 sqlite3/3.42.0
Found cached dataset parquet (/work3/s212722/herd/datasets/cache/alexrs___parquet/alexrs--alpaca-cleaned-10-clusters-e0d2361138237477/0.0.0/14a00e99c0d15a23649d0db8944380ac81082d4b021f398733dd84f3a6c569a7)
Loading cached processed dataset at /work3/s212722/herd/datasets/cache/alexrs___parquet/alexrs--alpaca-cleaned-10-clusters-e0d2361138237477/0.0.0/14a00e99c0d15a23649d0db8944380ac81082d4b021f398733dd84f3a6c569a7/cache-78d2e6d3debffc6e.arrow
2023-11-05 13:08:41.339 | INFO     | herd.finetune_utils:get_peft_config:30 - Fine-tuning with MoLoRA using 10 experts
Loading checkpoint shards:   0%|          | 0/2 [00:00<?, ?it/s]Loading checkpoint shards:  50%|‚ñà‚ñà‚ñà‚ñà‚ñà     | 1/2 [04:04<04:04, 244.49s/it]Loading checkpoint shards: 100%|‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà| 2/2 [05:27<00:00, 149.41s/it]Loading checkpoint shards: 100%|‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà‚ñà| 2/2 [05:27<00:00, 163.67s/it]
2023-11-05 13:14:24.237 | WARNING  | herd.finetune:_finetune_experts:200 - Number of experts is not 1 but we are training an expert, setting it to 1. num_experts: 1
/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/trl/trainer/utils.py:246: UserWarning: The passed formatting_func has more than one argument. Usually that function should have a single argument `example` which corresponds to the dictonnary returned by each element of the dataset. Make sure you know what you are doing.
  warnings.warn(
2023-11-05 13:14:39.786 | INFO     | herd.finetune:_finetune_experts:232 - Training expert: expert_6, output_dir: /work3/s212722/herd/meta-llama/Llama-2-7b-hf/alpaca-10/q_v_r4/molora/expert_6
wandb: Currently logged in as: alexrs95. Use `wandb login --relogin` to force relogin
wandb: wandb version 0.15.12 is available!  To upgrade, please run:
wandb:  $ pip install wandb --upgrade
wandb: Tracking run with wandb version 0.15.10
wandb: Run data is saved locally in /zhome/03/c/164482/code/herd/wandb/run-20231105_131441-71b25gtz
wandb: Run `wandb offline` to turn off syncing.
wandb: Syncing run molora/expert_6
wandb: ‚≠êÔ∏è View project at https://wandb.ai/alexrs95/herd-llama
wandb: üöÄ View run at https://wandb.ai/alexrs95/herd-llama/runs/71b25gtz
  0%|          | 0/386 [00:00<?, ?it/s]You're using a LlamaTokenizerFast tokenizer. Please note that with a fast tokenizer, using the `__call__` method is faster than using a method to encode the text followed by a call to the `pad` method to get a padded encoding.
`use_cache=True` is incompatible with gradient checkpointing. Setting `use_cache=False`...
  0%|          | 1/386 [00:24<2:38:26, 24.69s/it]  1%|          | 2/386 [00:35<1:45:15, 16.45s/it]  1%|          | 3/386 [00:46<1:28:15, 13.83s/it]  1%|          | 4/386 [00:56<1:20:14, 12.60s/it]  1%|‚ñè         | 5/386 [01:07<1:15:46, 11.93s/it]  2%|‚ñè         | 6/386 [01:18<1:13:04, 11.54s/it]  2%|‚ñè         | 7/386 [01:29<1:11:20, 11.29s/it]  2%|‚ñè         | 8/386 [01:39<1:10:11, 11.14s/it]  2%|‚ñè         | 9/386 [01:50<1:09:24, 11.05s/it]  3%|‚ñé         | 10/386 [02:01<1:08:51, 10.99s/it]                                                    3%|‚ñé         | 10/386 [02:01<1:08:51, 10.99s/it]  3%|‚ñé         | 11/386 [02:12<1:08:26, 10.95s/it]  3%|‚ñé         | 12/386 [02:23<1:08:08, 10.93s/it]  3%|‚ñé         | 13/386 [02:34<1:07:53, 10.92s/it]  4%|‚ñé         | 14/386 [02:45<1:07:40, 10.92s/it]  4%|‚ñç         | 15/386 [02:56<1:07:29, 10.91s/it]  4%|‚ñç         | 16/386 [03:07<1:07:18, 10.92s/it]  4%|‚ñç         | 17/386 [03:17<1:07:08, 10.92s/it]  5%|‚ñç         | 18/386 [03:28<1:06:58, 10.92s/it]  5%|‚ñç         | 19/386 [03:39<1:06:49, 10.92s/it]  5%|‚ñå         | 20/386 [03:50<1:06:38, 10.93s/it]                                                    5%|‚ñå         | 20/386 [03:50<1:06:38, 10.93s/it]  5%|‚ñå         | 21/386 [04:01<1:06:28, 10.93s/it]  6%|‚ñå         | 22/386 [04:12<1:06:18, 10.93s/it]  6%|‚ñå         | 23/386 [04:23<1:06:09, 10.93s/it]Traceback (most recent call last):
  File "/zhome/03/c/164482/code/herd/main.py", line 78, in <module>
    main()
  File "/zhome/03/c/164482/code/herd/main.py", line 55, in main
    finetune(
  File "/zhome/03/c/164482/code/herd/herd/finetune.py", line 266, in finetune
    _finetune_experts(dataset, tokenizer, config, experts, peft_strategy, use_base, model_values, path_values)
  File "/zhome/03/c/164482/code/herd/herd/finetune.py", line 233, in _finetune_experts
    trainer.train()
  File "/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/transformers/trainer.py", line 1591, in train
    return inner_training_loop(
           ^^^^^^^^^^^^^^^^^^^^
  File "/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/transformers/trainer.py", line 1999, in _inner_training_loop
    self._maybe_log_save_evaluate(tr_loss, model, trial, epoch, ignore_keys_for_eval)
  File "/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/transformers/trainer.py", line 2339, in _maybe_log_save_evaluate
    self._save_checkpoint(model, trial, metrics=metrics)
  File "/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/transformers/trainer.py", line 2396, in _save_checkpoint
    self.save_model(output_dir, _internal_call=True)
  File "/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/transformers/trainer.py", line 2901, in save_model
    self._save(output_dir)
  File "/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/transformers/trainer.py", line 2959, in _save
    self.model.save_pretrained(
  File "/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/peft/peft_model.py", line 180, in save_pretrained
    self.create_or_update_model_card(save_directory)
  File "/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/peft/peft_model.py", line 781, in create_or_update_model_card
    card.save(filename)
  File "/zhome/03/c/164482/code/herd/.venv/lib/python3.11/site-packages/huggingface_hub/repocard.py", line 131, in save
    with open(filepath, mode="w", newline="", encoding="utf-8") as f:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 122] Disk quota exceeded: '/work3/s212722/herd/meta-llama/Llama-2-7b-hf/alpaca-10/q_v_r4/molora/expert_6/checkpoint-23/README.md'
wandb: Waiting for W&B process to finish... (failed 1). Press Control-C to abort syncing.
wandb: 
wandb: Run history:
wandb:         train/epoch ‚ñÅ‚ñà
wandb:   train/global_step ‚ñÅ‚ñà
wandb: train/learning_rate ‚ñÅ‚ñÅ
wandb:          train/loss ‚ñà‚ñÅ
wandb: 
wandb: Run summary:
wandb:         train/epoch 0.1
wandb:   train/global_step 20
wandb: train/learning_rate 0.0003
wandb:          train/loss 0.891
wandb: 
wandb: üöÄ View run molora/expert_6 at: https://wandb.ai/alexrs95/herd-llama/runs/71b25gtz
wandb: Ô∏è‚ö° View job at https://wandb.ai/alexrs95/herd-llama/jobs/QXJ0aWZhY3RDb2xsZWN0aW9uOjk5NjEyNzE2/version_details/v14
wandb: Synced 5 W&B file(s), 0 media file(s), 0 artifact file(s) and 0 other file(s)
wandb: Find logs at: ./wandb/run-20231105_131441-71b25gtz/logs
